system/include/wasi/api.h

```diff
@@ -1008,6 +1008,19 @@ typedef uint32_t __wasi_lookupflags_t;
 _Static_assert(sizeof(__wasi_lookupflags_t) == 4, "witx calculated size");
 _Static_assert(_Alignof(__wasi_lookupflags_t) == 4, "witx calculated align");
 
+/**
+ * Default FD for the root folder.
+ * Why file descriptor 3?
+ * While not specified, the most common WASI implementation, wasi-libc, expects
+ * POSIX style file descriptor allocation, where the lowest available number is
+ * used to open the next file. Since 1 and 2 are taken by stdout and stderr,
+ * `root` is assigned 3.
+ *   - https://github.com/WebAssembly/WASI/issues/122
+ *   - https://pubs.opengroup.org/onlinepubs/9699919799/functions/V2_chap02.html#tag_15_14
+ *   - https://github.com/WebAssembly/wasi-libc/blob/wasi-sdk-16/libc-bottom-half/sources/preopens.c#L215
+ */
+#define __WASI_FD_ROOT 3
```

Comment:
I believe the way it works is that there can be any number of pre-opens, starting at 3: https://github.com/WebAssembly/wasi-libc/blob/30094b6ed05f19cee102115215863d185f2db4f0/libc-bottom-half/sources/preopens.c#L212-L215.   And non of these pre-opens are necessarily the root, they can be mounted in various places.

Also, can you move this macros (if you keep it), into standalone.c.   api.h is copied from wasi-libc, and I think its better to avoid have local modifications to it.